--- qtbase-opensource-src-5.9.0-beta/src/network/ssl/qsslcertificate_openssl.cpp.0058~	2017-04-08 13:48:57.377091332 +0200
+++ qtbase-opensource-src-5.9.0-beta/src/network/ssl/qsslcertificate_openssl.cpp	2017-04-08 13:48:57.420091630 +0200
@@ -1,6 +1,7 @@
 /****************************************************************************
 **
 ** Copyright (C) 2016 The Qt Company Ltd.
+** Copyright (C) 2016 Richard J. Moore <rich@kde.org>
 ** Contact: https://www.qt.io/licensing/
 **
 ** This file is part of the QtNetwork module of the Qt Toolkit.
--- qtbase-opensource-src-5.9.0-beta/src/network/ssl/qsslcontext_openssl.cpp.0058~	2017-04-08 13:48:57.346091117 +0200
+++ qtbase-opensource-src-5.9.0-beta/src/network/ssl/qsslcontext_openssl.cpp	2017-04-08 13:48:57.420091630 +0200
@@ -3,6 +3,7 @@
 ** Copyright (C) 2016 The Qt Company Ltd.
 ** Copyright (C) 2014 BlackBerry Limited. All rights reserved.
 ** Copyright (C) 2014 Governikus GmbH & Co. KG.
+** Copyright (C) 2016 Richard J. Moore <rich@kde.org>
 ** Contact: https://www.qt.io/licensing/
 **
 ** This file is part of the QtNetwork module of the Qt Toolkit.
--- qtbase-opensource-src-5.9.0-beta/src/network/ssl/qssldiffiehellmanparameters_openssl.cpp.0058~	2017-04-08 13:48:57.420091630 +0200
+++ qtbase-opensource-src-5.9.0-beta/src/network/ssl/qssldiffiehellmanparameters_openssl.cpp	2017-04-08 13:49:35.708358591 +0200
@@ -1,6 +1,7 @@
 /****************************************************************************
 **
 ** Copyright (C) 2015 Mikkel Krautz <mikkel@krautz.dk>
+** Copyright (C) 2016 Richard J. Moore <rich@kde.org>
 ** Contact: https://www.qt.io/licensing/
 **
 ** This file is part of the QtNetwork module of the Qt Toolkit.
--- qtbase-opensource-src-5.9.0-beta/src/network/ssl/qsslkey_openssl.cpp.0058~	2017-04-08 13:48:57.338091062 +0200
+++ qtbase-opensource-src-5.9.0-beta/src/network/ssl/qsslkey_openssl.cpp	2017-04-08 13:48:57.420091630 +0200
@@ -1,6 +1,7 @@
 /****************************************************************************
 **
 ** Copyright (C) 2016 The Qt Company Ltd.
+** Copyright (C) 2016 Richard J. Moore <rich@kde.org>
 ** Contact: https://www.qt.io/licensing/
 **
 ** This file is part of the QtNetwork module of the Qt Toolkit.
--- qtbase-opensource-src-5.9.0-beta/src/network/ssl/qsslsocket_openssl.cpp.0058~	2017-04-08 13:48:57.378091339 +0200
+++ qtbase-opensource-src-5.9.0-beta/src/network/ssl/qsslsocket_openssl.cpp	2017-04-08 13:48:57.420091630 +0200
@@ -2,6 +2,7 @@
 **
 ** Copyright (C) 2016 The Qt Company Ltd.
 ** Copyright (C) 2014 Governikus GmbH & Co. KG
+** Copyright (C) 2016 Richard J. Moore <rich@kde.org>
 ** Contact: https://www.qt.io/licensing/
 **
 ** This file is part of the QtNetwork module of the Qt Toolkit.
--- qtbase-opensource-src-5.9.0-beta/src/network/ssl/qsslsocket_openssl_symbols.cpp.0058~	2017-04-08 13:48:57.378091339 +0200
+++ qtbase-opensource-src-5.9.0-beta/src/network/ssl/qsslsocket_openssl_symbols.cpp	2017-04-08 13:48:57.421091637 +0200
@@ -2,6 +2,7 @@
 **
 ** Copyright (C) 2016 The Qt Company Ltd.
 ** Copyright (C) 2014 BlackBerry Limited. All rights reserved.
+** Copyright (C) 2016 Richard J. Moore <rich@kde.org>
 ** Contact: https://www.qt.io/licensing/
 **
 ** This file is part of the QtNetwork module of the Qt Toolkit.
--- qtbase-opensource-src-5.9.0-beta/src/network/ssl/qsslsocket_openssl_symbols_p.h.0058~	2017-04-08 13:48:57.378091339 +0200
+++ qtbase-opensource-src-5.9.0-beta/src/network/ssl/qsslsocket_openssl_symbols_p.h	2017-04-08 13:48:57.421091637 +0200
@@ -2,6 +2,7 @@
 **
 ** Copyright (C) 2016 The Qt Company Ltd.
 ** Copyright (C) 2014 BlackBerry Limited. All rights reserved.
+** Copyright (C) 2016 Richard J. Moore <rich@kde.org>
 ** Contact: https://www.qt.io/licensing/
 **
 ** This file is part of the QtNetwork module of the Qt Toolkit.
