diff -up qtbase-everywhere-src-5.15.0-beta1/src/plugins/sqldrivers/sqlite/qsql_sqlite.cpp.omv~ qtbase-everywhere-src-5.15.0-beta1/src/plugins/sqldrivers/sqlite/qsql_sqlite.cpp
--- qtbase-everywhere-src-5.15.0-beta1/src/plugins/sqldrivers/sqlite/qsql_sqlite.cpp.omv~	2020-03-10 03:41:07.648627699 +0100
+++ qtbase-everywhere-src-5.15.0-beta1/src/plugins/sqldrivers/sqlite/qsql_sqlite.cpp	2020-03-10 03:41:52.696626278 +0100
@@ -206,9 +206,9 @@ void QSQLiteResultPrivate::initColumns(b
         QString colName = QString(reinterpret_cast<const QChar *>(
                     sqlite3_column_name16(stmt, i))
                     ).remove(QLatin1Char('"'));
-        const QString tableName = QString(reinterpret_cast<const QChar *>(
+/*        const QString tableName = QString(reinterpret_cast<const QChar *>(
                             sqlite3_column_table_name16(stmt, i))
-                            ).remove(QLatin1Char('"'));
+                            ).remove(QLatin1Char('"')); */
         // must use typeName for resolving the type to match QSqliteDriver::record
         QString typeName = QString(reinterpret_cast<const QChar *>(
                     sqlite3_column_decltype16(stmt, i)));
@@ -241,7 +241,7 @@ void QSQLiteResultPrivate::initColumns(b
             }
         }
 
-        QSqlField fld(colName, fieldType, tableName);
+        QSqlField fld(colName, fieldType /*, tableName*/);
         fld.setSqlType(stp);
         rInf.append(fld);
     }
