diff -up qtbase-everywhere-src-5.14.0-beta1/src/corelib/io/qstandardpaths_unix.cpp.2~ qtbase-everywhere-src-5.14.0-beta1/src/corelib/io/qstandardpaths_unix.cpp
--- qtbase-everywhere-src-5.14.0-beta1/src/corelib/io/qstandardpaths_unix.cpp.2~	2019-10-09 16:49:55.538735597 +0200
+++ qtbase-everywhere-src-5.14.0-beta1/src/corelib/io/qstandardpaths_unix.cpp	2019-10-09 17:05:00.317442410 +0200
@@ -151,7 +151,10 @@ QString QStandardPaths::writableLocation
             xdgRuntimeDir = QDir::tempPath() + QLatin1String("/runtime-") + userName;
             fileInfo.setFile(xdgRuntimeDir);
             if (!fileInfo.isDir()) {
-                if (!QDir().mkdir(xdgRuntimeDir)) {
+                mode_t old_umask = umask(0077);
+                bool const ok = QDir().mkdir(xdgRuntimeDir);
+                umask(old_umask);
+                if (!ok) {
                     qErrnoWarning("QStandardPaths: error creating runtime directory %ls",
                                   qUtf16Printable(xdgRuntimeDir));
                     return QString();
@@ -175,10 +178,19 @@ QString QStandardPaths::writableLocation
         }
         // "The directory MUST be owned by the user"
         if (fileInfo.ownerId() != myUid) {
-            qWarning("QStandardPaths: wrong ownership on runtime directory %ls, %d instead of %d",
-                     qUtf16Printable(xdgRuntimeDir),
-                     fileInfo.ownerId(), myUid);
-            return QString();
+            // We're probably setuid with XDG_RUNTIME_DIR persisting. Let's do the right thing here...
+            xdgRuntimeDir = QFile::decodeName("/run/user/") + QString::number(myUid);
+            fileInfo.setFile(xdgRuntimeDir);
+            if(!fileInfo.isDir()) {
+                    mode_t old_umask = umask(0077);
+                    bool const ok = QDir().mkdir(xdgRuntimeDir);
+                    umask(old_umask);
+                    if(!ok) {
+                        qWarning("QStandardPaths: error creating runtime directory %s: %s", qPrintable(xdgRuntimeDir), qPrintable(qt_error_string(errno)));
+                        return QString();
+                    }
+            }
+            return xdgRuntimeDir;
         }
         // "and he MUST be the only one having read and write access to it. Its Unix access mode MUST be 0700."
         // since the current user is the owner, set both xxxUser and xxxOwner
