--- qtbase-opensource-src-5.8.0-beta/qmake/generators/unix/unixmake2.cpp.omv~	2016-11-07 22:21:28.964737064 +0100
+++ qtbase-opensource-src-5.8.0-beta/qmake/generators/unix/unixmake2.cpp	2016-11-07 22:24:06.382976494 +0100
@@ -180,7 +180,7 @@ UnixMakefileGenerator::writeMakeParts(QT
         const ProStringList &incs = project->values("INCLUDEPATH");
         for(int i = 0; i < incs.size(); ++i) {
             const ProString &inc = incs.at(i);
-            if (inc.isEmpty())
+            if (inc.isEmpty() || inc.toQString() == "/usr/include")
                 continue;
 
             if (!isystem.isEmpty() && isSystemInclude(inc.toQString()))
