diff -up qtbase-everywhere-src-5.15.1/qmake/generators/unix/unixmake2.cpp.4~ qtbase-everywhere-src-5.15.1/qmake/generators/unix/unixmake2.cpp
--- qtbase-everywhere-src-5.15.1/qmake/generators/unix/unixmake2.cpp.4~	2020-09-09 18:05:48.666478040 +0200
+++ qtbase-everywhere-src-5.15.1/qmake/generators/unix/unixmake2.cpp	2020-09-09 18:10:55.541102171 +0200
@@ -201,7 +201,7 @@ UnixMakefileGenerator::writeMakeParts(QT
         const ProStringList &incs = project->values("INCLUDEPATH");
         for(int i = 0; i < incs.size(); ++i) {
             const ProString &inc = incs.at(i);
-            if (inc.isEmpty())
+            if (inc.isEmpty() || inc.toQString() == "/usr/include")
                 continue;
 
             t << " -I" << escapeFilePath(inc);
