--- qtbase-opensource-src-5.9.0-alpha/config.tests/unix/pcre2/pcre2.cpp.omv~	2017-03-03 03:41:39.026287310 +0100
+++ qtbase-opensource-src-5.9.0-alpha/config.tests/unix/pcre2/pcre2.cpp	2017-03-03 03:41:43.803308050 +0100
@@ -39,7 +39,7 @@
 
 #include <pcre2.h>
 
-#if (PCRE2_MAJOR < 10) || ((PCRE2_MAJOR == 10) && (PCRE_MINOR < 20))
+#if (PCRE2_MAJOR < 10) || ((PCRE2_MAJOR == 10) && (PCRE2_MINOR < 20))
 #error This PCRE version is not supported
 #endif
 
