diff -Naur qtbase-opensource-src-5.6.0/config.tests/unix/journald/journald.pro qtbase-opensource-src-5.6.0.tpg/config.tests/unix/journald/journald.pro
--- qtbase-opensource-src-5.6.0/config.tests/unix/journald/journald.pro 2016-03-03 11:35:22.000000000 +0000
+++ qtbase-opensource-src-5.6.0.tpg/config.tests/unix/journald/journald.pro     2016-05-31 17:07:54.060642091 +0000
@@ -1,6 +1,10 @@
 SOURCES = journald.c
 
 CONFIG += link_pkgconfig
-PKGCONFIG_PRIVATE += libsystemd-journal
+
+packagesExist(libsystemd): \
+    PKGCONFIG_PRIVATE += libsystemd
+else: \
+    PKGCONFIG_PRIVATE += libsystemd-journal
 
 CONFIG -= qt
diff -Naur qtbase-opensource-src-5.6.0/src/corelib/global/global.pri qtbase-opensource-src-5.6.0.tpg/src/corelib/global/global.pri
--- qtbase-opensource-src-5.6.0/src/corelib/global/global.pri   2016-03-03 11:35:22.000000000 +0000
+++ qtbase-opensource-src-5.6.0.tpg/src/corelib/global/global.pri       2016-05-31 17:09:27.968602913 +0000
@@ -53,7 +53,10 @@
 
 journald {
     CONFIG += link_pkgconfig
-    PKGCONFIG_PRIVATE += libsystemd-journal
+    packagesExist(libsystemd): \
+        PKGCONFIG_PRIVATE += libsystemd
+    else: \
+        PKGCONFIG_PRIVATE += libsystemd-journal
     DEFINES += QT_USE_JOURNALD
 }
 
