--- qtbase-opensource-src-5.6.0-beta/config.tests/unix/alsa/alsatest.cpp.than	2015-12-14 21:49:46.000000000 +0100
+++ qtbase-opensource-src-5.6.0-beta/config.tests/unix/alsa/alsatest.cpp	2016-02-11 13:29:59.249275394 +0100
@@ -32,7 +32,7 @@
 ****************************************************************************/
 
 #include <alsa/asoundlib.h>
-#if(!(SND_LIB_MAJOR == 1 && SND_LIB_MINOR == 0 && SND_LIB_SUBMINOR >= 10))
+#if(!(SND_LIB_MAJOR == 1 && (SND_LIB_MINOR > 0 || SND_LIB_SUBMINOR >= 10)))
 #error "Alsa version found too old, require >= 1.0.10"
 #endif
 
