diff -up qtbase-everywhere-src-5.14.1/src/gui/text/qfontdatabase.cpp.omv~ qtbase-everywhere-src-5.14.1/src/gui/text/qfontdatabase.cpp
--- qtbase-everywhere-src-5.14.1/src/gui/text/qfontdatabase.cpp.omv~	2020-02-10 23:40:47.847949827 +0100
+++ qtbase-everywhere-src-5.14.1/src/gui/text/qfontdatabase.cpp	2020-02-10 23:41:37.977946545 +0100
@@ -194,14 +194,14 @@ struct QtFontStyle
         signed int  weight : 8;
         signed int stretch : 12;
 
-        bool operator==(const Key & other) {
+        bool operator==(const Key & other) const {
             return (style == other.style && weight == other.weight &&
                     (stretch == 0 || other.stretch == 0 || stretch == other.stretch));
         }
-        bool operator!=(const Key &other) {
+        bool operator!=(const Key &other) const {
             return !operator==(other);
         }
-        bool operator <(const Key &o) {
+        bool operator <(const Key &o) const {
             int x = (style << 12) + (weight << 14) + stretch;
             int y = (o.style << 12) + (o.weight << 14) + o.stretch;
             return (x < y);
diff -up qtbase-everywhere-src-5.14.1/src/widgets/dialogs/qwizard.cpp.omv~ qtbase-everywhere-src-5.14.1/src/widgets/dialogs/qwizard.cpp
--- qtbase-everywhere-src-5.14.1/src/widgets/dialogs/qwizard.cpp.omv~	2020-02-10 23:41:51.486945660 +0100
+++ qtbase-everywhere-src-5.14.1/src/widgets/dialogs/qwizard.cpp	2020-02-10 23:42:08.757944529 +0100
@@ -255,11 +255,11 @@ public:
     bool extension = false;
     bool sideWidget = false;
 
-    bool operator==(const QWizardLayoutInfo &other);
-    inline bool operator!=(const QWizardLayoutInfo &other) { return !operator==(other); }
+    bool operator==(const QWizardLayoutInfo &other) const;
+    inline bool operator!=(const QWizardLayoutInfo &other) const { return !operator==(other); }
 };
 
-bool QWizardLayoutInfo::operator==(const QWizardLayoutInfo &other)
+bool QWizardLayoutInfo::operator==(const QWizardLayoutInfo &other) const
 {
     return topLevelMarginLeft == other.topLevelMarginLeft
            && topLevelMarginRight == other.topLevelMarginRight
