--- qtbase-opensource-src-5.5.1/src/widgets/kernel/qwidget.cpp.omv~	2016-02-07 13:27:24.299956456 +0100
+++ qtbase-opensource-src-5.5.1/src/widgets/kernel/qwidget.cpp	2016-02-07 13:27:52.840218719 +0100
@@ -7149,7 +7149,7 @@ void QWidgetPrivate::setGeometry_sys(int
 
     bool needsShow = false;
 
-    if (q->isWindow()) {
+    if (q->isWindow() || q->windowHandle()) {
         if (!(data.window_state & Qt::WindowFullScreen) && (w == 0 || h == 0)) {
             q->setAttribute(Qt::WA_OutsideWSRange, true);
             if (q->isVisible() && q->testAttribute(Qt::WA_Mapped))
