diff -up qtbase-everywhere-src-5.14.0-beta1/src/xml/sax/qxml_p.h.1~ qtbase-everywhere-src-5.14.0-beta1/src/xml/sax/qxml_p.h
--- qtbase-everywhere-src-5.14.0-beta1/src/xml/sax/qxml_p.h.1~	2019-10-09 16:46:56.079397180 +0200
+++ qtbase-everywhere-src-5.14.0-beta1/src/xml/sax/qxml_p.h	2019-10-09 16:49:30.852689042 +0200
@@ -229,7 +229,7 @@ private:
     // for the DTD currently being parsed.
     static const int dtdRecursionLimit = 2;
     // The maximum amount of characters an entity value may contain, after expansion.
-    static const int entityCharacterLimit = 4096;
+    static const int entityCharacterLimit = 65536;
 
     const QString &string();
     void stringClear();
