diff -up qtbase-everywhere-src-5.15.1/src/gui/kernel/qguiapplication.cpp.7~ qtbase-everywhere-src-5.15.1/src/gui/kernel/qguiapplication.cpp
--- qtbase-everywhere-src-5.15.1/src/gui/kernel/qguiapplication.cpp.7~	2020-09-02 12:15:07.000000000 +0200
+++ qtbase-everywhere-src-5.15.1/src/gui/kernel/qguiapplication.cpp	2020-10-02 03:37:27.944409786 +0200
@@ -3162,11 +3162,13 @@ void QGuiApplicationPrivate::processScre
     bool availableGeometryChanged = e->availableGeometry != s->d_func()->availableGeometry;
     s->d_func()->availableGeometry = e->availableGeometry;
 
-    if (geometryChanged) {
-        Qt::ScreenOrientation primaryOrientation = s->primaryOrientation();
+    const Qt::ScreenOrientation primaryOrientation = s->primaryOrientation();
+    if (geometryChanged)
         s->d_func()->updatePrimaryOrientation();
 
-        emit s->geometryChanged(s->geometry());
+    s->d_func()->emitGeometryChangeSignals(geometryChanged, availableGeometryChanged);
+
+    if (geometryChanged) {
         emit s->physicalSizeChanged(s->physicalSize());
         emit s->physicalDotsPerInchChanged(s->physicalDotsPerInch());
         emit s->logicalDotsPerInchChanged(s->logicalDotsPerInch());
@@ -3178,8 +3180,6 @@ void QGuiApplicationPrivate::processScre
             updateFilteredScreenOrientation(s);
     }
 
-    s->d_func()->emitGeometryChangeSignals(geometryChanged, availableGeometryChanged);
-
     resetCachedDevicePixelRatio();
 }
 
diff -up qtbase-everywhere-src-5.15.1/src/gui/kernel/qscreen.cpp.7~ qtbase-everywhere-src-5.15.1/src/gui/kernel/qscreen.cpp
--- qtbase-everywhere-src-5.15.1/src/gui/kernel/qscreen.cpp.7~	2020-09-02 12:15:07.000000000 +0200
+++ qtbase-everywhere-src-5.15.1/src/gui/kernel/qscreen.cpp	2020-10-02 03:36:05.274088741 +0200
@@ -88,6 +88,9 @@ void QScreenPrivate::updateGeometriesWit
 void QScreenPrivate::emitGeometryChangeSignals(bool geometryChanged, bool availableGeometryChanged)
 {
     Q_Q(QScreen);
+    if (geometryChanged)
+        emit q->geometryChanged(geometry);
+
     if (availableGeometryChanged)
         emit q->availableGeometryChanged(availableGeometry);
 
